app/validation/validator.py

```diff
@@ -333,6 +337,9 @@ def validate_when_rule(when_clause, answer_ids_with_group_id, referenced_id):
                     return Validator._error_message('The "when" clause for {} contains a comparison_id and uses a '
                                                     'condition of unset or set'
                                                     .format(referenced_id))
+            if 'answer_ids' in when:
+                if len(when['answer_ids']) != len(set(when['answer_ids'])):
+                    return Validator._error_message('Duplicate answer ids found within {} clause'.format(referenced_id))
```

Comment:
I realise this is existing, but I can't see why it returns once it's found one error. The other validation methods build an array of errors and return the array.

Reply:
Fixed

Reply:
Fixed